index.js

```diff
@@ -0,0 +1,65 @@
+const tilesContainer = document.querySelector(".tiles");
+
+//memotest images
+
+const feelingsImgLink = {
+    happy: "images/feelings/happy.png",
+    sad: "images/feelings/sad.png", 
+    angry: "images/feelings/angry.png",
+    hungry: "images/feelings/hungry.png",
+    thirsty: "images/feelings/thirsty.png",
+    scared: "images/feelings/scared.png",
+}
+
+const feelingsImages = {}; //object that contains images to append
+const feelingsPairs = []; //this is to shuffle the cards internally
+
+for (const [feeling, src] of Object.entries(feelingsImgLink)) {
+const img = document.createElement("img");
+img.src = src;
+feelingsImages[feeling] = img;
+feelingsPairs.push(feeling, feeling); //this is to have each feeling twice in an array that will be shuffled later
+}
+
+//create each tile
+function buildTile (feeling) {
+    const card = document.createElement("img");
+    card.src = "images/square_backside_pattern.png"
+
+    card.classList.add("card");
+    card.setAttribute("data-feeling", feeling) // this is to know which feeling is asigned to the card, whether it is revealed or not
+    tilesContainer.appendChild(card); //maybe we need to get this out of the function & put it in the for loop
+
+    let unrevealedCard = true; //toggle between states
+    card.addEventListener("click", () => {
+        if (unrevealedCard) {
+            card.src = feelingsImgLink[feeling];
+            unrevealedCard = false;
+            
+        } else {
+            card.src = "images/square_backside_pattern.png";
+            unrevealedCard = true;
+            
+        }
+
+        //add animation no matter the click
+        card.classList.add("animation"); //i got this animation from a css animation builder 
+
+        //remove it so the class is added and the animation triggered every time there's a click. the delay time matches the duration of the animation (0.75s)
+        setTimeout(() => {
+            card.classList.remove("animation")
+        }, 750);
```

Comment:
this is really nice :) the animation will break though is someone is a nasty clicker and double clicks for example. From UX perspective I think it's fine, but there are some really nice tools you might want to look at as you are visibly quite advanced already. I would recommend taking look at a `gsap` library for example :)

Reply:
You're absolutely right and I didn't really like that. I'll take a look at the library you shared. I also think that appending the card image and changing its src depending on the click is not quite accurate. Because the image surely changes, but you can see the transition between one image and the other is quite abrupt. I added that specific animation because it makes the abrupt transition less noticeable. Now, I think I should've created an individual div and appended different images depending on the click. That way there's an exit animation and and entrance one to make the transition smoother.